From ae966bdc79adbb6600de6a59fd0d0b706b31541b Mon Sep 17 00:00:00 2001
From: Thomas Orozco <thomas@orozco.fr>
Date: Thu, 7 Apr 2016 12:45:30 +0200
Subject: [PATCH] Exit when unable to connect to Docker daemon

Practically speaking, we'd rather have no cAdvisor than have a cAdvisor
that doesn't report the metrics we care about. This change causes
cAdvisor to exit if it can't connect to the Docker daemon (it'll then
be restart by Upstart).
---
 manager/manager.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/manager/manager.go b/manager/manager.go
index fd14f6b..b30d12d 100644
--- a/manager/manager.go
+++ b/manager/manager.go
@@ -132,7 +132,7 @@ func New(memoryCache *memory.InMemoryCache, sysfs sysfs.SysFs, maxHousekeepingIn
 
 	dockerInfo, err := dockerInfo()
 	if err != nil {
-		glog.Warningf("Unable to connect to Docker: %v", err)
+		glog.Fatalf("Unable to connect to Docker: %v", err)
 	}
 	rktPath, err := rkt.RktPath()
 	if err != nil {
@@ -218,7 +218,7 @@ type manager struct {
 func (self *manager) Start() error {
 	err := docker.Register(self, self.fsInfo, self.ignoreMetrics)
 	if err != nil {
-		glog.Errorf("Docker container factory registration failed: %v.", err)
+		glog.Fatalf("Docker container factory registration failed: %v.", err)
 	}
 
 	err = rkt.Register(self, self.fsInfo, self.ignoreMetrics)
-- 
2.7.4

